crates/bevy_transform/src/components/transform2d.rs

```diff
@@ -0,0 +1,372 @@
+    }
+
+    /// Creates a new [`Transform2d`] with `translation`.
+    ///
+    /// Rotation will be `0.`, scale will be `Vec2::ONE` and `z_translation` will be `0.`.
+    #[inline]
+    pub const fn from_translation(translation: Vec2) -> Self {
+        Transform2d {
+            translation,
+            ..Self::IDENTITY
+        }
+    }
+
+    /// Creates a new [`Transform2d`] with `translation`.
+    ///
+    /// Rotation will be `0.` and scale will be `Vec2::ONE`
+    #[inline]
+    pub const fn from_translation_3d(Vec3 { x, y, z }: Vec3) -> Self {
+        Transform2d {
+            translation: Vec2 { x, y },
+            z_translation: z,
+            ..Self::IDENTITY
+        }
+    }
+
+    /// Creates a new [`Transform2d`] with `rotation`.
+    ///
+    /// Translation will be `Vec2::ZERO`, scale will be `Vec2::ONE` and `z_translation` will be `0.`.
+    #[inline]
+    pub const fn from_rotation(rotation: f32) -> Self {
+        Transform2d {
+            rotation,
+            ..Self::IDENTITY
+        }
+    }
+
+    /// Creates a new [`Transform2d`] with `scale`.
+    ///
+    /// Translation will be `Vec2::ZERO`, rotation will be `0.` and `z_translation` will be `0.`
+    #[inline] // Hmm const
+    pub const fn from_scale(scale: Vec2) -> Self {
+        Transform2d {
+            scale,
+            ..Self::IDENTITY
+        }
+    }
+
+    /// Returns this [`Transform2d`] with a new translation.
+    #[must_use]
+    #[inline]
+    pub const fn with_translation(mut self, translation: Vec2) -> Self {
+        self.translation = translation;
+        self
+    }
+
+    /// Returns this [`Transform2d`] with a new rotation.
+    #[must_use]
+    #[inline]
+    pub const fn with_rotation(mut self, rotation: f32) -> Self {
+        self.rotation = rotation;
+        self
+    }
+
+    /// Returns this [`Transform2d`] with a new scale.
+    #[must_use]
+    #[inline]
+    pub const fn with_scale(mut self, scale: Vec2) -> Self {
+        self.scale = scale;
+        self
+    }
+
+    /// Returns this [`Transform2d`] with a new Z translation.
+    #[must_use]
+    #[inline]
+    pub const fn with_z_translation(mut self, z_translation: f32) -> Self {
+        self.z_translation = z_translation;
+        self
+    }
+
+    /// Returns this [`Transform2d`] rotated so the local `direction` points in the given `target_direction`.
+    ///
+    /// # Example
+    /// ```
+    /// # use bevy_transform::prelude::*;
+    /// # use bevy_math::prelude::*;
+    /// // Create a transform rotated so that up(Vec2::Y) points to the right.
+    /// let transform = Transform2d::IDENTITY.pointed_to(Vec2::Y, Vec2::X);
+    ///
+    /// approx::assert_abs_diff_eq!(transform.up(), Vec2::X);
+    /// ```
+    ///
+    /// If this [`Transform2d`] has a parent, the `point` is relative to the [`Transform2d`] of the parent.
+    #[inline]
+    pub fn pointed_to(mut self, direction: Vec2, target_direction: Vec2) -> Self {
```

Comment:
"point to" and "point at" to me literally means the same thing. It took me reading the argument list to understand the difference. I suggest `pointed_in` instead of `pointed_to` (because you  can point "in a direction" but not "in a position").

Reply:
The `_to` suffix matches the existing `looking_to` method on `Transform`. I think we should match `Transform` as much as possible in this PR.

I do remember talking with @mockersf about the naming here. We might want to revisit that, but ideally not in this PR :)

Reply:
Opened #8452